File: solutions/typescript/0011.ts
function maxArea(height: number[]): number {
  let left = 0;
  let right = height.length - 1;
  let maxWater = 0;

  while (left < right) {
    // 计算当前容器的水量
    const width = right - left;
    const h = Math.min(height[left], height[right]);
    const currentWater = width * h;

    // 更新最大水量
    maxWater = Math.max(maxWater, currentWater);

    // 移动较短的线，因为较长的线可能与其他线形成更大的容器
    if (height[left] < height[right]) {
      left++;
    } else {
      right--;
    }
  }

  return maxWater;
}

File: solutions/rust/0011.rs
impl Solution {
    pub fn max_area(height: Vec<i32>) -> i32 {
        let mut left = 0;
        let mut right = height.len() - 1;
        let mut max_water = 0;

        while left < right {
            let width = (right - left) as i32;
            let h = height[left].min(height[right]);
            let current_water = width * h;

            max_water = max_water.max(current_water);

            if height[left] < height[right] {
                left += 1;
            } else {
                right -= 1;
            }
        }
        max_water
    }
}